skactiveml/pool/_core_set.py

```diff
@@ -0,0 +1,242 @@
+        unlabeled samples
+     selected_samples: np.ndarray of shape (n_selected_samples, )
+        index of datapoints already selects
+     batch_size: int, optional(default=1)
+        The number of samples to be selected in one AL cycle.
+     random_state: np.random.RandomState
+        The random state to use
+     missing_label: scalar or string or np.nan or None, default=np.nan
+        Value to represent a missing label
+     mapping: np.ndarray of shape (n_candidates, ) default None
+        Index array that maps `candidates` to `X`.
+        (`candidates = X[mapping]`)
+     n_new_cand: int or None, default None
+        The number of new candidates that are additionally added to X.
+        Only used for the case, that in the query function with the
+        shape of candidates is (n_candidates, n_feature)
+        
+     Return:
+     ----------
+     query_indices: numpy.ndarry of shape (batch_size, )
+         The query_indices indicate for which candidate sample a label is
+         to queried from the candidates.
+         If candidates in None or of shape (n_candidates), the indexing
+         refers to samples in X.
+         If candidates is of shape (n_candidates, n_features), the indexing
+         refers to samples in candidates.
+     utilities: numpy.ndarray of shape (batch_size, n_samples) or
+         numpy.ndarry of shape (batch_size, n_new_cand)
+         The distance between each data point and its nearest center that used
+         for selecting the next sample.
+         If candidates is None or of shape (n_candidates), the indexing
+         refers to samples in X.
+         If candidates is of shape (n_candidates, n_features), the indexing
+         refers to samples in candidates.
+        """
+    # read the labeled aka selected samples from the y vector
+    selected_samples = labeled_indices(y, missing_label=missing_label)
+    if mapping is None:
+        mapping = unlabeled_indices(y, missing_label=missing_label)
+    # initialize the utilities matrix with
+    if n_new_cand is None:
+        utilities = np.empty(shape=(batch_size, X.shape[0]))
+    else:
+        utilities = np.empty(shape=(batch_size, n_new_cand))
+
+    query_indices = np.array([], dtype=int)
+
+    for i in range(batch_size):
+        if n_new_cand is None:
+            utilities[i] = update_distances(X, selected_samples, mapping)
+        else:
+            update_dist = update_distances(X, selected_samples, mapping)
+            utilities[i] = update_dist[mapping]
+
+        # select index
+        idx = np.nanargmax(utilities[i])
+
+        if len(selected_samples) == 0:
+            idx = random_state.choice(mapping)
+            # because np.nanargmax always return the first occurrence is returned
+
+        query_indices = np.append(query_indices, [idx])
+        selected_samples = np.append(selected_samples, [idx])
+
+    return query_indices, utilities
+
+def update_distances(X, cluster_centers, mapping):
```

Comment:
Add an additional parameter, which contains already computed distances for the previous cluster centers.